src/lib/signature_lib/schnorr.ml

```diff
@@ -300,15 +304,15 @@ end
 
 open Snark_params
 
-module Message = struct
+module Message(Context : CONTEXT) = struct
```

Comment:
Is it necessary to parametrize the whole `Message` module with signature kind rather than having a `~signature_kind` as a non-optional parameter to all of the functions in the module?

Reply:
I am currently experimenting with many options to get a good outcome, this will almost certainly change. It would be best to hold off on leaving comments as nothing is ready at this time